chainstate/src/detail/chainstateref.rs

```diff
@@ -525,6 +886,44 @@ impl<'a, S: BlockchainStorageRead, O: OrphanBlocks> ChainstateRef<'a, S, O> {
     }
 }
 
+// Calc not burned tokens that were placed in OutputValue::TokenTransfer after partial burn
+fn get_change_amount(
+    tx: &Transaction,
+    burn_token_id: &H256,
+    block_id: &Id<Block>,
+) -> Result<Amount, CheckBlockTransactionsError> {
+    let change_amount = tx
+        .outputs()
+        .iter()
+        .filter_map(|x| match x.value() {
+            OutputValue::Coin(_) => None,
+            OutputValue::Asset(asset) => match asset {
+                AssetData::TokenTransferV1 { token_id, amount } => {
+                    if token_id == burn_token_id {
+                        Some(amount)
+                    } else {
+                        None
+                    }
+                }
+                AssetData::TokenIssuanceV1 {
+                    token_ticker: _,
+                    amount_to_issue: _,
+                    number_of_decimals: _,
+                    metadata_uri: _,
+                } => None,
+                AssetData::TokenBurnV1 {
+                    token_id: _,
+                    amount_to_burn: _,
+                } => None,
```

Comment:
@TheQuantumPhysicist probably wouldn't like that, but I would write it this way:
```rust
AssetData::TokenIssuanceV1 { .. } | AssetData::TokenBurnV1 { .. } => None,
```

Reply:
Yeah, I miss that, agree with you

Reply:
Using `..` should be avoided. I don't mind using `|`. The idea is to force you to handle any change you do to data types.

Reply:
Don't we always want to return `None` here even if new fields are added?.. I agree that it is very nice for the language to force handling of new enum variants, but I'm not so sure about fields.

Reply:
What if it's TransferV2?

Reply:
I'm not sure I understand the comment about the potential new enum variant. If [the new variant is added](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=3b142204c49cbcfc5698ab6fac09336f), then it only needs to be handled in the match expression. So `TokenIssuanceV1` and `TokenBurnV1` can stay as they are. Perhaps I'm missing something?.. Are you talking about the situation when we need to introduce a new transfer version which will require to change the logic in token issuance and token burn handling without introducing the corresponding V2 versions?

Reply:
OK, if you are really insisting I'm not going to complain any longer and will always follow this rule, but I think that it should be mentioned somewhere in the `CONTRIBUTING.md`.